presto-main/src/main/java/io/prestosql/sql/planner/plan/JoinNode.java

```diff
@@ -118,12 +118,6 @@ public JoinNode(
                     "%s join do not work with %s distribution type",
                     type,
                     distributionType.get());
-            // It does not make sense to PARTITION when there is nothing to partition on
-            checkArgument(
-                    !(distributionType.get() == PARTITIONED && criteria.isEmpty() && type != RIGHT && type != FULL),
-                    "Equi criteria are empty, so %s join should not have %s distribution type",
-                    type,
-                    distributionType.get());
         }
 
```

Comment:
According to https://github.com/trinodb/trino/pull/3245, maybe we should remove the following as well:
```java
import static io.prestosql.sql.planner.plan.JoinNode.DistributionType.PARTITIONED;
```

Reply:
oh ok, I'm going to remove this